lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUCloseTaskTest.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Driver.Task;
+using Org.Apache.REEF.IMRU.OnREEF.Driver;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Logging;
+using Xunit;
+using TraceLevel = System.Diagnostics.TraceLevel;
+
+namespace Org.Apache.REEF.Tests.Functional.IMRU
+{
+    /// <summary>
+    /// This is to test close event handler in IMRU tasks
+    /// The test provide IRunningTask, IFailedTask and ICOmpletedTask handlers so that to trigger close events and handle the 
+    /// failed tasks and completed tasks
+    /// </summary>
+    [Collection("FunctionalTests")]
+    public class IMRUCloseTaskTest : IMRUBrodcastReduceTestBase
+    {
+        private const string CompletedTaskMessage = "CompletedTaskMessage";
+        private const string FailTaskMessage = "FailTaskMessage";
+
+        /// <summary>
+        /// This test is for running in local runtime
+        /// It sends close event for all the running tasks.
+        /// For tasks that can close properly after receiving the close event, the driver will get ICompletedTask event. 
+        /// For tasks that cannot close properly after receiving the close event, they will throw exceptions
+        /// and the driver will get IFailedTask event. 
+        /// </summary>
+        [Fact]
+        public void TestTaskCloseOnLocalRuntime()
+        {
+            int chunkSize = 2;
+            int dims = 50;
+            int iterations = 200;
+            int mapperMemory = 5120;
+            int updateTaskMemory = 5120;
+            int numTasks = 4;
+            string testFolder = DefaultRuntimeFolder + TestId;
+            TestBroadCastAndReduce(false, numTasks, chunkSize, dims, iterations, mapperMemory, updateTaskMemory, testFolder);
+            int failedCount = GetMessageCount(FailTaskMessage, "driver", DriverStdout, testFolder);
+            int completedCount = GetMessageCount(CompletedTaskMessage, "driver", DriverStdout, testFolder);
```

Comment:
The way `GetMessageCount` is currently implemented these two lines would read the same log twice. I think it can be rewritten to accept 2 parameters, log lines and message, and these two calls to `GetMessageCount` replaced with single `ReadLogFile` followed by two `GetMessageCount`s.